### DISCUSSION TITLE
Authentication in a Next.js app using WPGraphQL Cors (problems with iPhone/iPad login)

### DISCUSSION THREAD

**Discussion by alexostar** [wp-graphql/wp-graphql#2401] [Q&A]
I'm working on a Next.js that includes authentication with [native WP cookies as described by Kellen Mace]( https://developers.wpengine.com/blog/headless-wordpress-authentication-native-cookies) at WPEngine. I tested the deployment and added the URL of the test site to the WPGraphQL Cors *Extend "Access-Control-Allow-Origin” header* settings. Everything works fine when I test the app in Crome on my MacBook and on a Android phone but when testing it on iPad/iPhone the login does not work.
I got it however working by turning off the default "Prevent Cross-Site Tracking" in Safari but that is not a long-term solution. Anybody who has a better idea?

**Comment by sw4d** [comment]
Hey @IceSwede, just trying to best understand the issue. 

Is the Cross-Site tracking issue happening after doing the steps outlined in the production section of the tutorial?

--------------------------------------------------------------------------------
How to Use in Production
In order to use this authentication method in production, you need to do the following:

WordPress Backend
Install and activate the same plugins mentioned in the WordPress Backend Setup section, above.
From the WordPress admin sidebar, go to Graphql > Settings and click the CORS Settings tab.
Check the checkboxes next to these options:
– Send site credentials
– Enable login mutation
– Enable logout mutation
In the Extend "Access-Control-Allow-Origin” header field, enter the URL of your decoupled frontend JS app and click the button to save your changes. 
--------------------------------------------------------------------------------
Thanks for the clarification.

**Reply by alexostar** [reply]
Hi @sw4d Stephen, thank you for looking into this. Although my hobby-app is far from ready, I deployed it temporarily [here](https://hrxziyj3mm6wv3gwcda6qacs0.js.wpenginepowered.com/) to test the authentication from various devices. And I have followed the steps on the WordPress backend that you list in your reply. I can create a new user on all devices that I have tested (Chrome on MacBook, Chrome on Android and Safari on iPad and iPhone but the login only works on the MacBook/Android (unless i turn of the default Prevent Cross-Site Tracking" in Safari)

**Comment by alexostar** [comment]
I got this to work by disabling Safari's *Prevent cross-site tradking*. That is of course not a solution for production but at least I understand now better what the problem is.
